test/jdk/java/awt/Frame/MegaIconTest/MegaIconTest.java

```diff
@@ -0,0 +1,271 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Canvas;
+import java.awt.Color;
+import java.awt.Dialog;
+import java.awt.Dimension;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.GridLayout;
+import java.awt.Image;
+import java.awt.Label;
+import java.awt.MediaTracker;
+import java.awt.Panel;
+import java.awt.Toolkit;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.image.ImageProducer;
+import java.net.URL;
+
+/*
+ * @test
+ * @bug 4175560
+ * @summary Test use of user-defined icons
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual MegaIconTest
+ */
+
+public class MegaIconTest {
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                Each of the buttons in the main window represents a test
+                of certain icon functionality - transparency/opacity, scaling etc.
+                Clicking on each button brings up a window displaying the graphic
+                that should appear in the corresponding icon.
+                Click on each button, minimize the resulting window, and check that
+                the icon is displayed as the test name indicates.
+                On Win32, icons should also be displayed correctly in the title bar.
+                If all the test pass, then this test passes, else fail.
+                """;
+
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(10)
+                .columns(35)
+                .testUI(MegaIconTest::initialize)
+                .build()
+                .awaitAndCheck();
+    }
+
+    public static Frame initialize() {
+        //Create the iconTestFrames and add to IconTestButtons
+        IconTestButtons itb = new IconTestButtons(new IconTestFrame[]{
+                new IconTestFrame("Opaque, Scaled Icon Test",
+                        "fight.gif"),
```

Comment:
"fight.gif" file is missing.

Reply:
Probably waiting for me to OK it off-line. I've now done so.

Reply:
Pushed it now.

Reply:
Does the test mean opaque icon background by "opaque icon"? If yes, then should we use one .gif with opaque background and another with transparent background. Currently both gif files - **dukeWave.gif and fight.gif** both have transparent backgrounds.